packages/contractkit/src/utils/tx-signing.ts

```diff
@@ -22,17 +29,65 @@ function getPrivateKeyWithout0xPrefix(privateKey: string) {
 
 export class CeloProvider extends PrivateKeyWalletSubprovider {
   private readonly celoPrivateKey: string // Always 0x prefixed
+  private readonly accountAddress: string // hex-encoded, lower case alphabets
   private _chainId: number | null = null
 
   constructor(privateKey: string) {
     // This won't accept a privateKey with 0x prefix and will call that an invalid key.
     super(getPrivateKeyWithout0xPrefix(privateKey))
     // Prefix 0x here or else the signed transaction produces dramatically different signer!!!
     this.celoPrivateKey = '0x' + getPrivateKeyWithout0xPrefix(privateKey)
+    this.accountAddress = generateAccountAddressFromPrivateKey(this.celoPrivateKey).toLowerCase()
+  }
+
+  public getAccounts(): string[] {
+    return [this.accountAddress]
+  }
+
+  // Note: This method is async since it over-rides the parent async method.
+  public async getAccountsAsync(): Promise<string[]> {
```

Comment:
why this?

Reply:
Not required, I added it for brevity. Since I added a `sync` version. I am going to replace this with just a comment, how about that?

Reply:
Also changes `getAccounts` to be `getAccount` why return a singleton array.